airbyte-cdk/python/unit_tests/sources/file_based/scenarios/scenario_builder.py

```diff
@@ -59,37 +57,26 @@ def __init__(
         self.expected_check_error = expected_check_error
         self.expected_discover_error = expected_discover_error
         self.expected_read_error = expected_read_error
-        self.source = InMemoryFilesSource(
-            files,
-            file_type,
-            availability_strategy,
-            discovery_policy,
-            validation_policies,
-            parsers,
-            stream_reader,
-            self.configured_catalog(SyncMode.incremental if incremental_scenario_config else SyncMode.full_refresh),
-            file_write_options,
-            cursor_cls,
-        )
         self.incremental_scenario_config = incremental_scenario_config
         self.validate()
 
     def validate(self) -> None:
         assert self.name
         if not self.expected_catalog:
             return
-        streams = {s["name"] for s in self.config["streams"]}
+        if self.expected_read_error or self.expected_check_error:
+            return
+        # Only verify the streams if no errors are expected
+        streams = set([s.name for s in self.source.streams(self.config)])
         expected_streams = {s["name"] for s in self.expected_catalog["streams"]}
         assert expected_streams <= streams
 
     def configured_catalog(self, sync_mode: SyncMode) -> Optional[Mapping[str, Any]]:
-        if not self.expected_catalog:
-            return None
         catalog: Mapping[str, Any] = {"streams": []}
-        for stream in self.expected_catalog["streams"]:
+        for stream in self.source.streams(self.config):
```

Comment:
create the configured catalog from the source's streams instead of from the expected catalog